### PR TITLE
Only set file id if it's unset

### PR DESCRIPTION
**Pull request by jvns**
Thanks for the great library! I noticed an issue where the inode numbers on my files were changing, like this:

```
$ ls .mnt/ -li
12899649424779661251 dr-xr-xr-x - bork tags
$ ls .mnt/tags/ -li
2554126253176081818 lr-xr-xr-x 54 bork vtest -> ../commits/ab/ab0a86afc8a176c1afae1faab8b489dcba76c40b
$ ls .mnt/ -li
15722733243342951127 dr-xr-xr-x - bork tags <-- inode number changed
```

I tracked it down to this line of code, where (if I understand correctly) I think the goal is to set an inode number if it's unset. But it was overriding the inode number that I'd already set in my `syscall.Stat_t`, which made things a little weird.

So I thought it might make more sense to only set the inode number if it's unset. This change fixed the issue I was having. I might have misunderstood the intent though.

Basically `attrs.Fileid` gets set on the previous line  ([here](https://github.com/willscott/go-nfs/blob/a91b8b3bf0fd76882bb2f2a2f3c1256405c03f15/file.go#L116)) and it seems weird to immediately overwrite that value

### CODE REVIEW

**Comment by willscott** [comment]
This looks like a rough edge left over from the inclusion of inodes in https://github.com/willscott/go-nfs/pull/88

It seems like the ultimate goal probably ought to be that there's fairly consistent behavior in Fileid's being passed back associated with files. Having a special case in `ReadDirPlus` is probably ultimately the wrong layer to do that.

It doesn't look like there's always a handle associated with a file path that we ask for `ToFileAttribute` against, but maybe there should be?

Ultimately, the use of the handle as the fileID is very much a hack, and probably not as stable as clients might expect it to be. My gut reaction is that it might be better to hash the full filename/path to generate this ID when it isn't provided by the underlying filesystem.

I'll see if I can propose a unified refactor for this later today.
